File: Week 2/6. Binary Sequences Generation Without Consecutive 11/binary_gen_without_cons_11.cpp
#include <iostream> 

using namespace std; 

void generateSequence(int size, string sequence = "", char prev = '0') {
    if (sequence.length() == size) {
        cout << sequence << endl;
        return;
    }
    
    generateSequence(size, sequence + '0', '0');
    
    if (prev != '1') {
        generateSequence(size, sequence + '1', '1');
    }
}

int main() 
{ 
    int size;
    cin >> size;
    
    string sequence = "";
    generateSequence(size, sequence, '0');
}

File: Week 2/6. Binary Sequences Generation Without Consecutive 11/binary_gen_without_cons_11.py
def generateSequence(size, sequence):
    length = len(sequence)
    
    if length == size:
        print(sequence)
        return
    
    if length == 0 or sequence[length - 1] == '0':
        generateSequence(size, sequence + '0')
        generateSequence(size, sequence + '1')
    else:
        generateSequence(size, sequence + '0')

size = int(input())
generateSequence(size, '')
